internal/autodetect/main.go

```diff
@@ -185,3 +187,45 @@ func (a *autoDetect) TargetAllocatorAvailability() (targetallocator.Availability
 func (a *autoDetect) FIPSEnabled(_ context.Context) bool {
 	return fips.IsFipsEnabled()
 }
+
+// ApplyAutoDetect attempts to automatically detect relevant information for this operator.
+func ApplyAutoDetect(autoDetect AutoDetect, c *config.Config, logger logr.Logger) error {
```

Comment:
Is there any reason not to make this a method on `AutoDetect`?

Reply:
Yes, though this is a more impactful change since we need all mocks to implement it. That might also change the way the interface works.

Reply:
Allright, I'm fine leaving it as-is in this PR.